pymc/distributions/discrete.py

```diff
@@ -394,6 +394,10 @@ def logcdf(value, p):
             p <= 1,
         )
 
+    def get_moment(value, size, p):
+        p = at.full(size, p)
+        return at.switch(p < 0.5, np.int64(0), np.int64(1))
```

Comment:
Maybe instead of forcing int64, we could rely on the op’s dtype?

Reply:
```suggestion
        return at.switch(p < 0.5, at.zeros_like(value), at.ones_like(value))
```